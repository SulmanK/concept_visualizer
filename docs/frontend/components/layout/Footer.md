# Footer Component

The `Footer` component displays the application footer with links, legal information, and branding.

## Overview

The footer appears at the bottom of every page in the application and provides secondary navigation, copyright information, and additional links. It maintains a consistent appearance across all pages.

## Usage

```tsx
import { Footer } from "components/layout/Footer";

function App() {
  return (
    <div className="app">
      <Header />
      <main>{/* Page content */}</main>
      <Footer />
    </div>
  );
}
```

## Props

| Prop        | Type     | Default | Description                                 |
| ----------- | -------- | ------- | ------------------------------------------- |
| `className` | `string` | `''`    | Additional CSS class to apply to the footer |

## Implementation Details

```tsx
import React from "react";
import { Box, Container, Typography, Link, Grid } from "@mui/material";
import { styled } from "@mui/material/styles";

const StyledFooter = styled("footer")(({ theme }) => ({
  backgroundColor: theme.palette.background.paper,
  padding: theme.spacing(6, 0),
  marginTop: "auto", // Pushes footer to bottom when page content is short
  borderTop: `1px solid ${theme.palette.divider}`,
}));

export function Footer({ className = "" }: { className?: string }) {
  const currentYear = new Date().getFullYear();

  return (
    <StyledFooter className={className}>
      <Container maxWidth="lg">
        <Grid container spacing={4} justifyContent="space-between">
          <Grid item xs={12} sm={4}>
            <Typography variant="h6" color="textPrimary" gutterBottom>
              Concept Visualizer
            </Typography>
            <Typography variant="body2" color="textSecondary">
              Generate visual concepts with AI-powered technology
            </Typography>
          </Grid>

          <Grid item xs={12} sm={4}>
            <Typography variant="h6" color="textPrimary" gutterBottom>
              Quick Links
            </Typography>
            <Box component="ul" sx={{ p: 0, listStyle: "none" }}>
              <li>
                <Link href="/" color="inherit">
                  Home
                </Link>
              </li>
              <li>
                <Link href="/concepts" color="inherit">
                  My Concepts
                </Link>
              </li>
              <li>
                <Link href="/help" color="inherit">
                  Help Center
                </Link>
              </li>
            </Box>
          </Grid>

          <Grid item xs={12} sm={4}>
            <Typography variant="h6" color="textPrimary" gutterBottom>
              Legal
            </Typography>
            <Box component="ul" sx={{ p: 0, listStyle: "none" }}>
              <li>
                <Link href="/privacy" color="inherit">
                  Privacy Policy
                </Link>
              </li>
              <li>
                <Link href="/terms" color="inherit">
                  Terms of Service
                </Link>
              </li>
            </Box>
          </Grid>
        </Grid>

        <Box mt={5}>
          <Typography variant="body2" color="textSecondary" align="center">
            Â© {currentYear} Concept Visualizer. All rights reserved.
          </Typography>
        </Box>
      </Container>
    </StyledFooter>
  );
}
```

## Features

- Responsive design that adapts to different screen sizes
- Organized link sections for easy navigation
- Automatically updates the copyright year
- Uses Material-UI components for consistent styling
- Sticky footer that stays at the bottom even with minimal page content

## Accessibility

The Footer component is designed with accessibility in mind:

- Proper heading hierarchy for screen readers
- Semantic HTML elements
- Sufficient color contrast for text readability
- Proper link labeling for navigation

## Customization

The Footer can be customized through:

1. **CSS overrides**: Using the `className` prop
2. **Theme customization**: Changing the Material-UI theme variables
3. **Component extension**: Creating a wrapped version with additional links or content
