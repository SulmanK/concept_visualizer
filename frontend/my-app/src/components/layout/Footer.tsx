import React from 'react';
import { Link } from 'react-router-dom';

export interface FooterProps {
  /**
   * Current year - defaults to current year
   */
  year?: number;
}

/**
 * Application footer with navigation, links and copyright
 * Uses dark indigo color scheme with responsive two-column layout
 */
export const Footer: React.FC<FooterProps> = ({ 
  year = new Date().getFullYear() 
}) => {
  return (
    <footer className="w-full bg-indigo-900 text-indigo-100 py-12 mt-12 shadow-lg relative overflow-hidden">
      <div className="container max-w-7xl mx-auto px-6 lg:px-8">
        {/* Two-column grid layout */}
        <div className="grid grid-cols-1 md:grid-cols-3 gap-8 lg:gap-12">
          {/* Brand column - takes up 1/3 of the space */}
          <div className="flex flex-col md:col-span-1">
            <div className="flex items-center">
              <div className="w-10 h-10 bg-gradient-to-r from-primary to-primary-dark rounded-lg flex items-center justify-center text-white font-bold shadow-md">
                CV
              </div>
              <h2 className="ml-3 text-xl font-bold text-white">
                Concept Visualizer
              </h2>
            </div>
            
            <p className="mt-4 text-sm text-indigo-200 leading-relaxed">
              Create and refine visual concepts with AI. Generate unique logos, color schemes, and design assets for your brand and projects.
            </p>
            
            <div className="mt-6">
              <a href="https://jigsawstack.com/?ref=powered-by" rel="follow" className="block">
                <img
                  src="https://jigsawstack.com/badge.svg"
                  alt="Powered by JigsawStack. The One API for your next big thing."
                  className="h-10 max-w-full"
                />
              </a>
            </div>
          </div>
          
          {/* Links grid - takes up 2/3 of the space */}
          <div className="grid grid-cols-1 sm:grid-cols-2 gap-8 md:col-span-2">
            {/* Features column */}
            <div>
              <h3 className="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700">
                Features
              </h3>
              <ul className="space-y-2">
                <li>
                  <Link 
                    to="/create" 
                    className="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                  >
                    <svg className="w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 6v6m0 0v6m0-6h6m-6 0H6" />
                    </svg>
                    <span className="relative">
                      Create Concepts
                      <span className="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"></span>
                    </span>
                  </Link>
                </li>
                <li>
                  <Link 
                    to="/recent" 
                    className="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                  >
                    <svg className="w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
                    </svg>
                    <span className="relative">
                      Recent Concepts
                      <span className="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"></span>
                    </span>
                  </Link>
                </li>
                <li>
                  <Link 
                    to="/refine" 
                    className="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                  >
                    <svg className="w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z" />
                    </svg>
                    <span className="relative">
                      Refine Concepts
                      <span className="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"></span>
                    </span>
                  </Link>
                </li>
              </ul>
            </div>
            
            {/* Resources column */}
            <div>
              <h3 className="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700">
                Resources
              </h3>
              <ul className="space-y-2">
                <li>
                  <a
                    href="https://github.com/SulmanK/concept-visualizer"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                  >
                    <svg className="w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
                      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path>
                    </svg>
                    <span className="relative">
                      GitHub Repository
                      <span className="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"></span>
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href="https://jigsawstack.com"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                  >
                    <svg className="w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M13 10V3L4 14h7v7l9-11h-7z" />
                    </svg>
                    <span className="relative">
                      JigsawStack API
                      <span className="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"></span>
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        
        {/* Footer bottom with copyright and GitHub link */}
        <div className="mt-10 pt-6 border-t border-indigo-700 flex flex-col sm:flex-row justify-between items-center space-y-4 sm:space-y-0">
          <div className="text-xs text-indigo-300">
            &copy; {year} Concept Visualizer. All rights reserved.
          </div>
          
          <a
            href="https://github.com/SulmanK/concept-visualizer"
            className="w-8 h-8 flex items-center justify-center rounded-full bg-indigo-800/30 text-indigo-200 hover:bg-indigo-700/50 hover:text-white transition-all transform hover:-translate-y-1 hover:shadow-md"
            aria-label="GitHub"
            target="_blank"
            rel="noopener noreferrer"
          >
            <svg className="w-4 h-4" fill="currentColor" viewBox="0 0 24 24" aria-hidden="true">
              <path fillRule="evenodd" d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z" clipRule="evenodd" />
            </svg>
          </a>
        </div>
      </div>
    </footer>
  );
}; 

// Add default export to fix import issue
export default Footer;