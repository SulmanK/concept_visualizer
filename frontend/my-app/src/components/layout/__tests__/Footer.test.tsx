import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Footer } from '../Footer';

// Helper function to render with router
const renderWithRouter = (ui: React.ReactElement) => {
  return render(
    <MemoryRouter>
      {ui}
    </MemoryRouter>
  );
};

describe('Footer Component', () => {
  // Basic rendering tests
  test('renders footer with logo and title', () => {
    renderWithRouter(<Footer />);
    
    // Check for logo text
    const logo = screen.getByText('CV');
    expect(logo).toBeInTheDocument();
    
    // Check for title
    const title = screen.getByText('Concept Visualizer');
    expect(title).toBeInTheDocument();
  });
  
  test('renders tagline', () => {
    renderWithRouter(<Footer />);
    
    const tagline = screen.getByText('Create and refine visual concepts with AI');
    expect(tagline).toBeInTheDocument();
  });
  
  // Features section tests
  test('renders features section with links', () => {
    renderWithRouter(<Footer />);
    
    // Check for section title
    const featuresHeading = screen.getByText('Features');
    expect(featuresHeading).toBeInTheDocument();
    
    // Check for feature links
    const createLink = screen.getByText('Create Concepts');
    const refineLink = screen.getByText('Refine Concepts');
    
    expect(createLink).toBeInTheDocument();
    expect(refineLink).toBeInTheDocument();
  });
  
  // Resources section tests
  test('renders resources section with links', () => {
    renderWithRouter(<Footer />);
    
    // Check for section title
    const resourcesHeading = screen.getByText('Resources');
    expect(resourcesHeading).toBeInTheDocument();
    
    // Check for resource links
    const githubLink = screen.getByText('GitHub Repository');
    const jigsawStackLink = screen.getByText('JigsawStack API');
    
    expect(githubLink).toBeInTheDocument();
    expect(jigsawStackLink).toBeInTheDocument();
  });
  
  // Copyright test
  test('renders copyright with current year', () => {
    // Mock current year
    const currentYear = new Date().getFullYear();
    renderWithRouter(<Footer />);
    
    const copyright = screen.getByText(`© ${currentYear} Concept Visualizer. All rights reserved.`);
    expect(copyright).toBeInTheDocument();
  });
  
  test('renders copyright with custom year', () => {
    renderWithRouter(<Footer year={2025} />);
    
    const copyright = screen.getByText('© 2025 Concept Visualizer. All rights reserved.');
    expect(copyright).toBeInTheDocument();
  });
  
  // Snapshot tests
  describe('Snapshots', () => {
    test('default footer snapshot', () => {
      const { container } = renderWithRouter(<Footer />);
      expect(container.firstChild).toMatchSnapshot();
    });
    
    test('footer with custom year snapshot', () => {
      const { container } = renderWithRouter(<Footer year={2025} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
}); 