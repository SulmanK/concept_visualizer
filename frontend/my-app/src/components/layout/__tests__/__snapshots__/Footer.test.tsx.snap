// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > Snapshots > default footer snapshot 1`] = `
<footer
  class="w-full bg-indigo-900 text-indigo-100 py-8 sm:py-12 mt-8 sm:mt-12 shadow-lg relative overflow-hidden"
>
  <div
    class="container max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-8 lg:gap-12"
    >
      <div
        class="flex flex-col md:col-span-1"
      >
        <div
          class="flex items-center justify-center md:justify-start"
        >
          <div
            class="w-10 h-10 bg-gradient-to-r from-primary to-primary-dark rounded-lg flex items-center justify-center text-white font-bold shadow-md"
          >
            CV
          </div>
          <h2
            class="ml-3 text-xl font-bold text-white"
          >
            Concept Visualizer
          </h2>
        </div>
        <p
          class="mt-4 text-sm text-indigo-200 leading-relaxed text-center md:text-left"
        >
          Create and refine visual concepts with AI. Generate unique logos, color schemes, and design assets for your brand and projects.
        </p>
        <div
          class="mt-6 flex justify-center md:justify-start"
        >
          <a
            class="block"
            href="https://jigsawstack.com/?ref=powered-by"
            rel="follow"
          >
            <img
              alt="Powered by JigsawStack. The One API for your next big thing."
              class="h-10 max-w-full"
              src="https://jigsawstack.com/badge.svg"
            />
          </a>
        </div>
      </div>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 gap-8 md:col-span-2"
      >
        <div>
          <h3
            class="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700 text-center sm:text-left"
          >
            Features
          </h3>
          <ul
            class="space-y-3 flex flex-col items-center sm:items-start"
          >
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/create"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 6v6m0 0v6m0-6h6m-6 0H6"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Create Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/recent"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Recent Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/refine"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Refine Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700 text-center sm:text-left"
          >
            Resources
          </h3>
          <ul
            class="space-y-3 flex flex-col items-center sm:items-start"
          >
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="https://github.com/SulmanK/concept-visualizer"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                  />
                </svg>
                <span
                  class="relative"
                >
                  GitHub Repository
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="https://jigsawstack.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 10V3L4 14h7v7l9-11h-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  JigsawStack API
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="mt-8 sm:mt-10 pt-4 sm:pt-6 border-t border-indigo-700 flex flex-col sm:flex-row justify-between items-center space-y-4 sm:space-y-0"
    >
      <div
        class="text-xs text-indigo-300 text-center sm:text-left"
      >
        Â©
        2025
         Concept Visualizer. All rights reserved.
      </div>
      <a
        aria-label="GitHub"
        class="w-8 h-8 flex items-center justify-center rounded-full bg-indigo-800/30 text-indigo-200 hover:bg-indigo-700/50 hover:text-white transition-all transform hover:-translate-y-1 hover:shadow-md"
        href="https://github.com/SulmanK/concept-visualizer"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="w-4 h-4"
          fill="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            clip-rule="evenodd"
            d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</footer>
`;

exports[`Footer Component > Snapshots > footer with custom year snapshot 1`] = `
<footer
  class="w-full bg-indigo-900 text-indigo-100 py-8 sm:py-12 mt-8 sm:mt-12 shadow-lg relative overflow-hidden"
>
  <div
    class="container max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-8 lg:gap-12"
    >
      <div
        class="flex flex-col md:col-span-1"
      >
        <div
          class="flex items-center justify-center md:justify-start"
        >
          <div
            class="w-10 h-10 bg-gradient-to-r from-primary to-primary-dark rounded-lg flex items-center justify-center text-white font-bold shadow-md"
          >
            CV
          </div>
          <h2
            class="ml-3 text-xl font-bold text-white"
          >
            Concept Visualizer
          </h2>
        </div>
        <p
          class="mt-4 text-sm text-indigo-200 leading-relaxed text-center md:text-left"
        >
          Create and refine visual concepts with AI. Generate unique logos, color schemes, and design assets for your brand and projects.
        </p>
        <div
          class="mt-6 flex justify-center md:justify-start"
        >
          <a
            class="block"
            href="https://jigsawstack.com/?ref=powered-by"
            rel="follow"
          >
            <img
              alt="Powered by JigsawStack. The One API for your next big thing."
              class="h-10 max-w-full"
              src="https://jigsawstack.com/badge.svg"
            />
          </a>
        </div>
      </div>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 gap-8 md:col-span-2"
      >
        <div>
          <h3
            class="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700 text-center sm:text-left"
          >
            Features
          </h3>
          <ul
            class="space-y-3 flex flex-col items-center sm:items-start"
          >
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/create"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 6v6m0 0v6m0-6h6m-6 0H6"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Create Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/recent"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Recent Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="/refine"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.232 5.232l3.536 3.536m-2.036-5.036a2.5 2.5 0 113.536 3.536L6.5 21.036H3v-3.572L16.732 3.732z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  Refine Concepts
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="text-sm font-semibold uppercase tracking-wider text-white pb-2 mb-3 border-b border-indigo-700 text-center sm:text-left"
          >
            Resources
          </h3>
          <ul
            class="space-y-3 flex flex-col items-center sm:items-start"
          >
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="https://github.com/SulmanK/concept-visualizer"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                  />
                </svg>
                <span
                  class="relative"
                >
                  GitHub Repository
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-sm text-indigo-200 hover:text-white transition-colors duration-200 flex items-center group"
                href="https://jigsawstack.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="w-4 h-4 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 10V3L4 14h7v7l9-11h-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="relative"
                >
                  JigsawStack API
                  <span
                    class="absolute bottom-0 left-0 w-0 h-px bg-white transition-all duration-200 group-hover:w-full"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="mt-8 sm:mt-10 pt-4 sm:pt-6 border-t border-indigo-700 flex flex-col sm:flex-row justify-between items-center space-y-4 sm:space-y-0"
    >
      <div
        class="text-xs text-indigo-300 text-center sm:text-left"
      >
        Â©
        2025
         Concept Visualizer. All rights reserved.
      </div>
      <a
        aria-label="GitHub"
        class="w-8 h-8 flex items-center justify-center rounded-full bg-indigo-800/30 text-indigo-200 hover:bg-indigo-700/50 hover:text-white transition-all transform hover:-translate-y-1 hover:shadow-md"
        href="https://github.com/SulmanK/concept-visualizer"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="w-4 h-4"
          fill="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            clip-rule="evenodd"
            d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</footer>
`;
